backend/models/issue.js

```diff
@@ -191,6 +188,99 @@ class Issue extends Ticket {
 		return await super.update(attributeBlacklist, user, sessionId, account, model, issueId, data, this.onBeforeUpdate.bind(this));
 	}
 
+	async getBCF(account, model, branch, revId, ids, useIssueNumbers = false) {
+		const projection = {};
+		const noClean = true;
+
+		const settings = await ModelSetting.findById({account, model}, model);
+		const issues = await this.findByModelName(account, model, branch, revId, projection, ids, noClean, useIssueNumbers);
+
+		return BCF.getBCFZipReadStream(account, model, issues, settings.properties.unit);
+	}
+
+	async importBCF(requester, account, model, revId, zipPath) {
+		const settings = await ModelSetting.findById({account, model}, model);
+
+		return BCF.importBCF(requester, account, model, zipPath, settings).then((bcfIssues) => {
+			return this.merge(account, model, revId, bcfIssues, requester.socketId, requester.user);
+		});
+	}
+
+	async merge(account, model, revId, data, sessionId, user) {
+		let branch;
+
+		if (!revId) {
+			branch = "master";
+		}
+
+		const history = await History.getHistory({ account, model }, branch, revId, {_id: 1});
+
+		if (!history) {
+			return Promise.reject(responseCodes.MODEL_HISTORY_NOT_FOUND);
```

Comment:
I would do this check prior BCF import, we don't really want to waste all this time to do the parsing then to realise it's all for nothing!